qiskit/providers/ibmq/ibmqjob.py

```diff
@@ -271,8 +285,7 @@ def status(self):
         elif 'ERROR' in api_job['status']:
             # Error status are of the form "ERROR_*_JOB"
             self._status = JobStatus.ERROR
-            # TODO: This seems to be an inconsistency in the API package.
-            self._api_error_msg = api_job.get('error') or api_job.get('Error')
+            self._api_error_msg = self._get_error_details()
```

Comment:
This would end up implying a call to `api.get_job()`, and fetching the results, right? One of the constraints introduced previously was that that `job.status()` should explicitly _not_ consume results (ie. only call `api.get_status_job()`, as in some premium configurations, the results can only be read once, and are deleted after the first read).

For consistency and avoiding that scenario (even if errored jobs might behave slightly different), can you move the call to `job.error_message()` or a similar solution that deals with "read-once" results - perhaps populating `self._api_error_msg` only the first time it is called or a similar measure, etc.